### DISCUSSION TITLE
PocketBase Framework Setup Guide

### DISCUSSION THREAD

**Discussion by TokenChingy** [pocketbase/pocketbase#5957] [Show and tell]
Hi all!

I recently wrote a short blog post on using PocketBase as a framework. It covers everything from the initial setup to integrating plugins and enforcing configurations in a CI/CD-friendly way. I plan to continue the series to cover all sorts of things from using combined JS and Go hooks, custom SQL drivers, cron jobs, etc.

[Check it out!](https://jasonlei.com/mastering-pocketbase-building-a-flexible-backend-framework-for-saas-and-mobile-apps)

Let me know your thoughts, and happy coding! 🎉

**Comment by dschissler** [comment]
I'd like to see Pocketbase be able to offer a wrapper for much of that `examples/base.main.go` setup code that is flexible enough to be used in most projects.

**Reply by TokenChingy** [reply]
It would be nice, but also, it's pretty straightforward to copy-paste what you would need from the example file. Although for most Go implementations, you wouldn't need the GitHub Update functionality (I would assume).
